src/devices/Arduino/FirmataCommandSequence.cs

```diff
@@ -11,26 +12,117 @@ namespace Iot.Device.Arduino
     /// A firmata command sequence
     /// Intended to be changed to public visibility later
     /// </summary>
-    internal class FirmataCommandSequence
+    public class FirmataCommandSequence
     {
+        /// <summary>
+        /// Start of sysex command byte. Used as start byte for almost all extended commands.
+        /// </summary>
+        public const byte StartSysex = (byte)FirmataCommand.START_SYSEX;
+
+        /// <summary>
+        /// End of sysex command byte. Must end all sysex commands.
+        /// </summary>
+        public const byte EndSysex = (byte)FirmataCommand.END_SYSEX;
+
         private List<byte> _sequence;
 
         /// <summary>
         /// Create a new command sequence
         /// </summary>
         /// <param name="command">The first byte of the command</param>
-        public FirmataCommandSequence(FirmataCommand command = FirmataCommand.START_SYSEX)
+        internal FirmataCommandSequence(FirmataCommand command = FirmataCommand.START_SYSEX)
         {
             _sequence = new List<byte>()
             {
                 (byte)command
             };
         }
 
-        internal List<byte> Sequence => _sequence;
+        /// <summary>
+        /// Create a new sysex command sequence. The <see cref="StartSysex"/> byte is added automatically.
+        /// </summary>
+        public FirmataCommandSequence()
+            : this(FirmataCommand.START_SYSEX)
+        {
+        }
 
+        /// <summary>
+        /// The current sequence
+        /// </summary>
+        public IReadOnlyList<byte> Sequence => _sequence;
+
+        /// <summary>
+        /// The current length of the sequence
+        /// </summary>
         public int Length => _sequence.Count;
 
+        /// <summary>
+        /// Decode an uint from packed 7-bit data.
+        /// This way of encoding uints is only used in extension modules.
+        /// </summary>
+        /// <param name="data">Data. 5 bytes expected</param>
+        /// <param name="fromOffset">Start offset in data</param>
+        /// <returns>The decoded unsigned integer</returns>
+        /// <exception cref="InvalidDataException">The received data is invalid</exception>
+        public static UInt32 DecodeUInt32(ReadOnlySpan<byte> data, int fromOffset)
+        {
+            for (int i = 0; i < 5; i++)
+            {
+                // Bit 7 of the data must always be 0, or there's either a communication problem or a protocol mismatch
+                if ((data[fromOffset + i] & 0x80) != 0)
+                {
+                    throw new InvalidDataException("An invalid byte was received. The message was probably corrupted");
+                }
+            }
+
+            Int32 value = data[fromOffset];
+            value |= data[fromOffset + 1] << 7;
```

Comment:
Instead of making the conversion yourself, why not use BitConverter.ToUInt32()?

Reply:
Because it's not the same. This converts an uint32 to 5x7 bits. The firmata protocol requires that all message payloads have the topmost bit set to 0, because bytes with the top bit set have special meanings (such as an end-of-message marker).